### DISCUSSION TITLE
Where does code store its data ?

### DISCUSSION THREAD

**Discussion by kabeza** [elementary/code#1361] [Q&A] [answered]
Does anyone know where does code store its config options, data, etc. ?

I've corrupted my ElementaryOS install, could access safe mode and connect to smb drive outside my PC

Now I'd like to copy/backup code's data folder so when I reinstall Elementary I restore that folder and then I could open code and have all the previous opened files and also the the unsaved one restored, right ?

Thanks

**Comment by jeremypw** [comment, marked as answer]
If you create a new file in Code it is saved temporarily in the `/home/<username>/.local/share/io.elementary.code/unsaved` folder until you save it yourself in the location of your choosing.

Once you have a document saved in the location of your choice, often a subdirectory of the `Documents` subfolder of your home folder (`/home/<username>/`), then that is where it stays until you move it.  While you are editing it, there is a backup file in the same folder with the same name plus a `~`.  If you have the "Save files when changed" option on, which I think is the default, then the document is automatically saved whenever you stop typing for more than 1 second.  So the disk file should differ very little from the document shown in Code.

Code's settings are stored in your home folder in the file `/home/<username>/.config/dconf/user` in the schema `io/elementary/code`.  This file needs the `dconf-editor` app to be read.

**Comment by kabeza** [comment]
Wow. Excellent answer @jeremypw 
Thanks
